### ISSUE TITLE
What is wrong with +gd option?

### ISSUE THREAD

**Issue by kherge** [phpbrew/phpbrew#461]
Using phpbrew on Ubuntu 14.04 with latest patches.

- The tool successfully installs PHP 5.6.4 but the `gd` extension is not listed in `php -m`, and I made sure that I properly switched to it before running the command and use the `+gd` variant when installing.
- I checked to see if I just needed to `phpbrew ext enable gd`, but it says that it is not installed.
- Doing a `phpbrew ext install gd` is successful (after using `--with-freetype-dir=/usr/include/freetype2`), but running PHP produces this error: `gd.so: undefined symbol: gdImageCreateFromWebp`

Is this a bug with phpbrew or the PHP build system?

**Comment by kherge** [comment]
This affects all versions of PHP that I try to install, but I wrote this little script that seems to work on all installation attempts:

```
phpbrew ext install gd -- \
 --enable-gd-native-ttf \
 --enable-wddx \
 --with-gd=shared \
 --with-freetype-dir=/usr/include/freetype2/freetype \
 --with-jpeg-dir=/usr \
 --with-libdir=lib/x86_64-linux-gnu \
 --with-png-dir=/usr \
 --with-vpx-dir=/usr \
 --with-xpm-dir=/usr
```

Not sure if this is something that can be integrated with phpbrew.

**Comment by marcioAlmada** [comment]
> I checked to see if I just needed to phpbrew ext enable gd, but it says that it is not installed.

If you want to be able to enable|disable an extension don't compile PHP with it. Use `phpbrew ext install <extname>` later. That's why I don't encourage `+everything`, most people regret it when they want to disable/enable extensions later :) (hopefully it's not your case)

> running PHP produces this error: gd.so: undefined symbol [...] Is this a bug with phpbrew or the PHP build system?

Build system is probably ok, I bet on a phpbrew bug or a missing requirement. Does it happens with all the PHPs?

**Comment by marcioAlmada** [comment]
> This affects all versions of PHP that I try to install

Oh, you posted while I was typing the reply. The solution you pointed out seems Ubuntu specific, I've just used `+gd` option successfully with Fedora and OSx, for instance.

**Comment by kherge** [comment]
Yeah, I figured this might be Ubuntu specific.

If this isn't something that can (or should) be handled by phpbrew, this issue can be closed but I recommend adding some information to the wiki about this problem.

**Comment by marcioAlmada** [comment]
Let's keep this one open. I broke my ubuntu VM some days ago while updating it but I'll try to investigate a fix after setup it again :)

**Comment by c9s** [comment]
FYI, I was using this one and it works for me on Mac OS:

    phpbrew ext install gd -- --with-gd-native--ttf --with-jpeg-dir=/opt/local --with-png-dir=/opt/local --with-zlib-dir=/opt/local --with-freetype-dir=/opt/local

**Comment by akomm** [comment]
I have a small not, platform independend, script for ubuntu (14.04 lts, but might be same on 12.x etc). I use to install gd with all usual formats supported, after switching to the desired php version using phpbrew:

```
#!/usr/bin/env bash

GNU_LIB=/usr/lib/x86_64-linux-gnu
phpbrew ext install gd -- --with-gd=shared --with-png-dir=$GNU_LIB --with-jpeg-dir=$GNU_LIB --with-freetype-dir=$GNU_LIB --enable-gd-native-ttf
```

It works perfectly well in my case.

**Comment by marcioAlmada** [comment]
This worked for me against all versions php versions (5.3+ to 7.0 alpha). A great addition to wiki ^^

**Comment by webmozart** [comment]
The command mentioned by @kherge worked for me. The option mentioned by @akomm did not. I'm on Ubuntu 14.04 LTS.

**Comment by Ronmi** [comment]
Add `--with-libpath=lib` or `--with-libpath=lib/x86_64-linux-gnu` might do the magic.

The `configure` script in `ext/gd` detects libraries by something like `test -f $PREFIX/$LIBPATH/libxxx.a`, where `$PREFIX` is  what you passed in `--with-xxxx-dir` and `$LIBPATH` can varies in different OS. By adding `--with-libpath`, you can set it up properly.

But not all extensions can built by adding this. The way `configure` detecting library also varies in different extensions, weired.

**Comment by mmenozzi** [comment]
Hi,
I have the same issue on Ubuntu 14.04 LTS trying to install gd extension with PHP 7.0.4.
I already tried all commands stated above.
Anyone can confirm?

**Comment by c9s** [comment]
@mmenozzi did you install gd as extension or install with gd?  could you provide your command and the build.log?

**Comment by c9s** [comment]
Note: here is the config used on Ubuntu
```
--with-gd=shared,/usr --enable-gd-native-ttf --with-gmp=shared,/usr --with-jpeg-dir=shared,/usr --with-xpm-dir=shared,/usr/X11R6 --with-png-dir=shared,/usr --with-freetype-dir=shared,/usr --with-vpx-dir=shared,/usr
```

**Comment by amansilla** [comment]
@c9s solution worked for me on Ubuntu 14.04.5 LTS 64bit

**Comment by mmenozzi** [comment]
@c9s thank you! Your solution worked for me!

**Comment by MurKit** [comment]
It does not work for me:

$ phpbrew ext install gd --with-gd=shared,/usr --enable-gd-native-ttf --with-gmp=shared,/usr --with-jpeg-dir=shared,/usr --with-xpm-dir=shared,/usr/X11R6 --with-png-dir=shared,/usr --with-freetype-dir=shared,/usr --with-vpx-dir=shared,/usr
===> Installing gd extension...
Log stored at: /home/machine/.phpbrew/build/php-5.3.24/ext/gd/build.log
Changing directory to /home/machine/.phpbrew/build/php-5.3.24/ext/gd
===> Phpize...
sh: 1: cannot create /home/machine/.phpbrew/build/php-5.3.24/ext/gd/build.log: Permission denied
Error: Command failed: phpize > /home/machine/.phpbrew/build/php-5.3.24/ext/gd/build.log 2>&1 returns:

$ sudo phpbrew ext install gd --with-gd=shared,/usr --enable-gd-native-ttf --with-gmp=shared,/usr --with-jpeg-dir=shared,/usr --with-xpm-dir=shared,/usr/X11R6 --with-png-dir=shared,/usr --with-freetype-dir=shared,/usr --with-vpx-dir=shared,/usr
Error: PHPBREW_PHP environment variable is not defined.
  This extension command requires you specify a PHP version from your build list.
  And it looks like you haven't switched to a version from the builds that were built with PHPBrew.
Suggestion: Please install at least one PHP with your prefered version and switch to it.

**Comment by sam452** [comment]
I can confirm @marcioAlmada approach worked for me. I installed php-5.6.32 with all my extensions except +gd. Then used @akomm command to install gd separately.

**Comment by ucay** [comment]
use @akomm's command worked for me. 
- Ubuntu 16.04
- PHP 7.1 
- System Wide Environment

**Comment by mzsolti** [comment]
In ubuntu 16.04 this command worked to install gd with phpbrew:
phpbrew ext install gd -- --with-gd=shared,/usr --enable-gd-native-ttf --with-gmp=shared,/usr --with-jpeg-dir=shared,/usr --with-xpm-dir=shared,/usr/X11R6 --with-png-dir=shared,/usr --with-freetype-dir=shared,/usr --with-vpx-dir=shared,/usr

**Comment by baldpale** [comment]
@mzsolti Thanks to this I could finally fix issue: Debian Jessie, PHP 7.0.17
